### ISSUE TITLE
Question about new network validation on Android

### ISSUE THREAD

**Issue by KrasnayaPloshchad** [jedisct1/dnscrypt-proxy#505]
On V2EX I have found a discussion for Android 7.1.1, where I found this version reformed network validation mechanism for WiFi connection, does DNSCrypt affect it?
See: https://www.v2ex.com/t/325630

**Comment by jedisct1** [comment]
Unfortunately, my Mandarin Chinese is not so hot. Can you summarize what the concern is about?

Captive portals either hijack HTTP queries, or DNS queries, or both. This makes them fundamentally incompatible with systems such as DNSCrypt that are designed to prevent hijacked responses from reaching your applications.

**Comment by KrasnayaPloshchad** [comment]
OK I have a chance now, so let me translate.
As their discussion, this is used to judge whether you have to use additional validation on web browser when you connected a WiFi and give you a notice. But considering several WiFi hotspots may use a LAN connecting for this validation, I concerned DNSCrypt would making this validating failed.

> Android 7.1.1 的网络验证改了

Android 7.1.1’s network validating have been changed.

> 多了个 /generate_204
> 看来需要在 http204.sinaapp.com 作说明了，这改动还是挺大的

Added `/generate_204`. It seems to me there is necessary to notice on http204.sinaapp.com, this is such a big change.

>请教一下，一直很好奇这个 204 服务是干嘛的？ iPhone 用户没接触过，在 V2 上看到了好多次
>
>iOS 上也有, 比如你到了麦丹劳连上了麦当劳的 wifi, iOS 能判断出这个 WIFI 需不需要再经过一次 web 验证, 其实跟这个一个原理. 在 Android 上, 这个地址本来是连到 Google 服务器上的, 因为墙的原因自然就连不上了, 所以得替换成国内的地址.

I’ve a question to consult you, (I) still curioused what does the 204 service do? iPhone user have not used, but seen it many times on V2 (V2EX).
iOS has it too. For example, in you go to McDonald you connect their WiFi, iOS can judge whether this WiFi need additional web validating, in fact there is the same mechanism on Android, this site in intended to connecting Google’s server, but unvailable because of GFW, so you have to connect domestic site.

>哦哦，是不是连接一个公共 wifi 之后，系统会访问这个地址来检验是否需要 web 验证？如果成功返回 204 则表示不需要？
>
>这是一个功能， 另外一个基本功能就是判断你现在链接的 wifi 到底有没有连上互联网，没的话就会有个小图标

Oooh, would the system access this site to check whether this need web validating after connecting a public WiFi? If successed then give back 204 to indicate unecessary?
This is just a function, another basic fonction is to judge whether the Wifi you connected is connecting the Internet, if not then give you an icon.

**Comment by jedisct1** [comment]
Captive portals often require disabling DNSCrypt, but you can reenable it after the portal is passed.
